icubam/predicu/plot/__init__.py

```diff
@@ -103,74 +113,60 @@ def plot_int(
     PLOTS.append(plot_name)
 
 
-def plot(plot_name, data, **plot_args):
+def plot(
+  plot_name: str,
+  cached_data: Dict[str, pd.DataFrame],
+  output_dir: str = DEFAULTS["output_dir"],
+  output_type: str = DEFAULTS["output_type"],
+  api_key: str = DEFAULTS["api_key"],
+  matplotlib_style: str = DEFAULTS["matplotlib_style"],
+):
   plot_module = __import__(f"{plot_name}", globals(), locals(), ["plot"], 1)
   plot_fun = plot_module.plot
   data_source = plot_module.data_source
+  cached_data[data_source] = load_if_not_cached(
+    data_source, cached_data, api_key=api_key
+  )
   matplotlib.use("agg")
-  matplotlib.style.use(plot_args["matplotlib_style"])
-  fig, tikzplotlib_kwargs = plot_fun(data=data[data_source].copy())
-  output_type = plot_args["output_type"]
-  if plot_args["output_type"] == "tex":
-    output_path = os.path.join(plot_args["output_dir"], f"{plot_name}.tex")
+  matplotlib.style.use(matplotlib_style)
+  fig, tikzplotlib_kwargs = plot_fun(data=cached_data[data_source].copy())
+  if output_type == "tex":
+    output_path = os.path.join(output_dir, f"{plot_name}.tex")
     __import__("tikzplotlib").save(
       filepath=output_path,
       figure=fig,
       **tikzplotlib_kwargs,
       standalone=True,
     )
   elif output_type in ["png", "pdf"]:
-    fig.savefig(
-      os.path.join(plot_args["output_dir"], f"{plot_name}.{output_type}")
-    )
+    fig.savefig(os.path.join(output_dir, f"{plot_name}.{output_type}"))
```

Comment:
For future refacorning, let's use, [pathlib](https://docs.python.org/3/library/pathlib.html) instead of os.path. Here it would have been,
```py
from pathlib import Path

def ...
   ...
   output_dir = Path(output_dir)
   ...
   fig.savefig(str(output_dir / f"{plot_name}.{output_type}")))
```

Reply:
Here, is fine, I'm not asking for changes in this PR.  I just mean that scripts that use a lot of paths manipulation are significantly better off using pathlib than os.path https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/ It helps readability for one thing.

Reply:
Very convincing article. I'll use that from now on :)